File: Recursion/exercises/js/fib.js
export default function fib(input) {
  if (input === 0) {
    return 0;
  }

  let arr = [0, 1];

  let helper = () => {
    if (input === arr.length - 1) {
      return;
    }

    arr.push(arr[arr.length - 1] + arr[arr.length - 2]);
    helper();
  };

  helper();

  return arr[arr.length - 1];
}

File: Recursion/exercises/go/fib/fib.go
package fib

func fib(input int) int {
	if input == 0 {
		return 0
	}

	fibSequence := make([]int, 1, input)
	fibSequence = append(fibSequence, 1)

	var helperFunc func()
	helperFunc = func() {
		if input == len(fibSequence)-1 {
			return
		}

		fibSequence = append(fibSequence, fibSequence[len(fibSequence)-1]+fibSequence[len(fibSequence)-2])
		helperFunc()
	}
	helperFunc()

	return fibSequence[len(fibSequence)-1]
}
